### ISSUE TITLE
Files attached to a content are not deleted after removing the image or deleting the content

### ISSUE THREAD

**Issue by marc-farre** [humhub/humhub#7108]
### What steps will reproduce the problem?

Create a post and attach a file, or add an image directly into the post.
Delete the file/image, or delete post.

### What is the expected result?

The file is removed from the `uploads/file/...` directory

### What do you get instead?

The file is still present.

### Additional info

| Q                | A
| ---------------- | ---
| HumHub version   | 1.16.1

**Comment by jvies** [comment]
On my side, the issue happens with Humhub 1.15.6 with files module 0.16.6.

**Comment by jvies** [comment]
Files now have effectively disappeared. Does it need a run of cron jobs ?

**Comment by marc-farre** [comment]
@jvies you're right, I've done more tests and updated the issue, because, except in the situation described above, the files are deleted after deleting the post or removing a file attached, after executing the following cron commands:

```sh
php yii cron/daily
php yii queue/run -v
```

**Comment by luke-** [comment]
This is true, currently files that are later removed from the rich text are not deleted.
This only happens as soon as the entire content (e.g. post) has been deleted.

For updated RichText e.g. `RichText::postProcess` should also handle this.

However, we must ensure that files that have been mapped to another place  (e.g. by a second richtext element of the content) are not unintentionally deleted.
